src/main/kotlin/kotlinx/kover/appliers/KoverMergedApplier.kt

```diff
@@ -169,10 +175,13 @@ private inline fun Project.mergedFilesProvider(
     extensionByProject: Provider<Map<Project, KoverProjectConfig>>,
     crossinline filterExtractor: (KoverProjectConfig) -> KoverSourceSetFilter
 ): Provider<Map<String, ProjectFiles>> {
-    return provider {
-        extensionByProject.get()
-            .map { (project, extension) -> project.path to project.projectFiles(filterExtractor(extension), extension) }
-            .associate { it }
+    return extensionByProject.map { extension ->
+        extension.map { (project, extension) ->
+            project.path to project.projectFiles(
+                filterExtractor(extension),
+                extension
+            )
+        }.associate { it }
     }
```

Comment:
using `providerX.map {}` (rather than `provider{ providerX.get()... }` means that if the provider is associated with a task, then that association remains. It is of course easy to confuse with Kotlin's `Collection.map {}` function...